medusa/cassandra_utils.py

```diff
@@ -238,45 +239,61 @@ def saved_caches_directory(self):
 
     @property
     def listen_address(self):
-        if 'listen_address' in self._config:
-            if self._config['listen_address']:
-                return self._config['listen_address']
+        if 'listen_address' in self._config and self._config['listen_address']:
+            return self._config['listen_address']
 
         return socket.gethostbyname(socket.getfqdn())
 
     @property
     def storage_port(self):
-        if 'storage_port' in self._config:
-            if self._config['storage_port']:
-                return self._config['storage_port']
-        return "7000"
+        if 'server_encryption_options' in self._config and \
+                self._config['server_encryption_options']['internode_encryption'] != "none" and \
+                'ssl_storage_port' in self._config and self._config['ssl_storage_port']:
+            return self._config['ssl_storage_port']
+
+        if 'storage_port' in self._config and self._config['storage_port']:
+            return self._config['storage_port']
+
+        raise CassandraConfigInvalidError('Unable to locate storage port setting.')
 
     @property
     def native_port(self):
-        if 'native_transport_port' in self._config:
-            if self._config['native_transport_port']:
-                return self._config['native_transport_port']
-        return "9042"
+
+        # Condition for client encryption enabled.
+        # When setting native_transport_port_ssl, expecting that non-encrypted will still be over existing
+        # native_transport_port. The encrypted will be over the native_transport_port_ssl.
+        # When not setting an alternate native_transport_port_ssl , the encrypted will be over
+        # the existing native_transport_port
+
+        if 'client_encryption_options' in self._config and \
+                self._config['client_encryption_options']['enabled'] == "true" \
+                and 'native_transport_port_ssl' in self._config \
+                and self._config['native_transport_port_ssl']:
+            return self._config['native_transport_port_ssl']
```

Comment:
You've removed the defaults here and in the storage port. Did you have a specific reason for this?
There's a case where the settings may not be present in the `cassandra.yaml` file and will then fallback to the defaults. Unless I'm missing something, we're not covering this case here.

My expectation would be to : 

- Check if encryption is enabled
- if so:
  - check if the SSL port is defined and return it, or return the default (9142)
- if not:
  - check if the non-SSL native port is defined and return it, or return the default (9042)

Same goes for the storage port by checking internode encryption.

Reply:
Gotcha, I see your point here.  I was thinking that if not specified, we were requiring a port designation in the c* config to avoid a health check on a port they removed.  I will remove the exception raised for that condition and use the default port literal values.